cmd/helm/dependency_build.go

```diff
@@ -65,7 +67,11 @@ func newDependencyBuildCmd(out io.Writer) *cobra.Command {
 			if client.Verify {
 				man.Verify = downloader.VerifyIfPossible
 			}
-			return man.Build()
+			err := man.Build()
+			if e, ok := err.(downloader.ErrRepoNotFound); ok {
+				return errors.Errorf("no repository definition for %s. Please add the missing repos via 'helm repo add'", strings.Join(e.Repos, ", "))
```

Comment:
couldn't this display repositories that might've succeeded? If I'm reading the code correctly, `e.Repos` and `missing` might not evaluate to the same result. There may only be a few repositories missing from the list.

I'd suggest appending the helpful error message to the output of `e.Error()`, like so:

```
return fmt.Errorf("%s. Please add the missing repos via 'helm repo add'", e.Error())
```

That way we leave the output of the error message up to the download manager.

Reply:
> couldn't this display repositories that might've succeeded? 

`Manager` method `hasAllRepos` ensures that all of the referenced deps are in the local repo cache and it doesn't return succeeded repositories. 

> If I'm reading the code correctly, `e.Repos` and `missing` might not evaluate to the same result. There may only be a few repositories missing from the list.

My bad! I did not process erros in the function `resolveRepoNames`，but i didn't find the reason why `e.Repos` and `missing` is different. Do you mean to merge `resolveRepoNames` with `hasAllRepos`?🤣

> 
> I'd suggest appending the helpful error message to the output of `e.Error()`, like so:
> 
> ```
> return fmt.Errorf("%s. Please add the missing repos via 'helm repo add'", e.Error())
> ```

Done; I removed some tips of `resolveRepoName` and is it necessary to show it here by a error handling function.

https://github.com/helm/helm/blob/241785c70fb38b2c074d7b3ddf0925812fb3fc69/pkg/downloader/manager.go#L469-L478

There seems to be an obvious error in this code